assets/js/components/GoogleChart.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * GoogleChart component.
+ *
+ * Site Kit by Google, Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState, useRef, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ProgressBar from './ProgressBar';
+
+let chartLoadPromise;
+
+async function loadCharts() {
+	if ( chartLoadPromise ) {
+		return chartLoadPromise;
+	}
+
+	// Inject the script if not already loaded and resolve on load.
+	if ( ! global.google || ! global.google.charts ) {
+		const script = document.createElement( 'script' );
+		script.type = 'text/javascript';
+
+		chartLoadPromise = new Promise( ( resolve ) => {
+			script.onload = resolve;
+			// Add the script to the DOM
+			global.document.head.appendChild( script );
+			// Set the `src` to begin transport
+			script.src = 'https://www.gstatic.com/charts/loader.js';
+		} );
+	} else {
+		// Charts is already available - resolve immediately.
+		chartLoadPromise = Promise.resolve();
+	}
+
+	return chartLoadPromise;
+}
+
+export default function GoogleChart( props ) {
+	const {
+		chartType,
+		className,
+		data,
+		loadCompressed,
+		loadHeight,
+		loadSmall,
+		loadText,
+		options,
+		selectedStats,
+		singleStat,
+	} = props;
+
+	const chartRef = useRef( null );
+	const [ chart, setChart ] = useState( null );
+	const [ loading, setLoading ] = useState( true );
+
+	const drawChart = useCallback( () => {
+		let dataTable = global.google?.visualization?.arrayToDataTable?.( data );
+		if ( dataTable ) {
+			if ( selectedStats.length > 0 ) {
+				dataTable = new global.google.visualization.DataView( dataTable );
+				if ( ! singleStat ) {
+					dataTable.setColumns( [
+						0,
+						...selectedStats.map( ( stat ) => stat + 1 ),
+					] );
+				}
+			}
+
+			chart.draw( dataTable, options );
+		}
```

Comment:
The definition here should move per my previous comments but I find it a little difficult to follow as written since it's mixing data tables and views (even though `chart.draw` accepts both) – for example only `DataView` has the `setColumns` method. How about this instead?
```suggestion
		let dataTable = global.google?.visualization?.arrayToDataTable?.( data );
		if ( ! dataTable ) {
			return;
		}
		if ( selectedStats.length > 0 ) {
			const dataView = new global.google.visualization.DataView( dataTable );
			if ( ! singleStat ) {
				dataView.setColumns(
					[ 0, ...selectedStats.map( ( stat ) => stat + 1 ) ]
				);
			}
			dataTable = dataView;
		}
		chart.draw( dataTable, options );
```